src/celeritas/phys/Interaction.hh

```diff
@@ -35,6 +35,7 @@ struct Interaction
     {
         scattered,  //!< Still alive, state has changed
         absorbed,  //!< Absorbed or transformed to another particle type
+        decay,  //!< N-body decay
```

Comment:
If we're going to shoehorn the decay process into the regular "interactor", this is exactly equivalent to absorption, right? So let's just delete this and the `from_decay`.

And in fact, you *need* to do this, because the `InteractionApplier` tries to change the direction if
`(result.action != Interaction::Action::absorbed)`, and it only kills the particle if it's absorbed.

So we should probably just delete this and use "from_absorption" . If you (or anyone else) think it's too confusing to mark it as `absorbed`, then just use
```suggestion
        decay = absorbed,  //!< N-body decay
```
and replace your function below with
```c++
CELER_FORCEINLINE_FUNCTION Interaction from_decay() { return from_absorption(); }

Reply:
But we still end up with `result.action == absorbed` in this case, right? Because having it tagged as decay in our actions is important. For example: in a perfect vacuum, a particle will never undergo absorption, but it will always decay if not stable.

Reply:
Currently we don't store the interaction result in any way or provide a way for the user to interrogate it. It's only used to update the post-step data in the InteractionApplier. I think that's OK, since we should be able to use the post-step action ID for users to interrogate that sort of thing.

Reply:
Maybe we should change the name from `absorbed` to `killed`? But then that might be ambiguous with a particle that had an error and had to be killed.

Reply:
So, what is then the downside or the necessity of introducing this additional Action, decay or from_decay()?  Indeed, it is very confusing that `decay = absorbed` as the decay process is a material independent (in a sense that the particle never interacts with a material and nerved be absorbed, but decay) even though minimizing the number of actions may be better if it simplifies underlying logic without loss of generality.

Reply:
Sorry that I used a wrong account.

Reply:
My initial enum addition was a starting point to expand the `InteractionApplier` when the decay process class comes in, as well as its inclusion in the stepping loop on future PRs. So we would have a particle killed if it is absorbed or underwent decay. For now, our actions are internal, but I assume that at some point we will be able to provide access to actions to the end user, similar to querying processes in G4, for analysis purposes. So one could, for example, store only data from a given decay, if their focus is, say, understanding signals from a given channel.

Reply:
Fair point. Should we rename `absorbed` to a catch-all word, such as `stopped`? This would cover all cases: scatters, stops (absorption, decay), remains unchanged or fails during sampling.

Reply:
`stopped` means "lost all energy but still exists" in the geant4 parlance right? We don't want that. Let's discuss at today's meeting.